lms/djangoapps/learning_success/management/commands/export_all_activity_records.py

```diff
@@ -304,32 +304,44 @@ class Command(BaseCommand):
 
     def add_arguments(self, parser):
         parser.add_argument('source_platform', type=str)
-        parser.add_argument('program_code', type=str)
+        parser.add_argument('programme_id', type=str)
 
-    def handle(self, source_platform, program_code, **kwargs):
+    def handle(self, source_platform, programme_id, **kwargs):
         """POST the collected data to the api endpoint from the settings
             Arguments:
                 source_platform: Platform import as, i.e. 'juniper' or 'ginkgo'
                 program_code: Program code of program to use 'disd'
         """
 
-        program = get_program_by_program_code(program_code)
-        student_data = list(all_student_data(program))
+        programme = get_program_by_program_code(programme_id)
+        student_data = list(all_student_data(programme))
 
         engine = create_engine(CONNECTION_STRING, echo=False)
         with engine.begin() as conn:
-            # remove existing
-            conn.execute("delete from lms_activity where source_platform = %s", (source_platform,))
-
+            # remove any existing rows for the day if exists
```

Comment:
Could you please add more info about this, either inline, or as a link to external documentation?

I understand that `engine.begin()` should utilise a transaction behind the scenes, so that's good, but still, I'd be looking to better understand the semantics of multiple POSTs in the same day - would that just be considered a scheduling configuration error that should be smoothed over?

Reply:
@yoniLavi The thought was to have one entry per day. With the delete statement we could easily re-run the extract on any day manually if we needed to which would then just replace the row without any further intervention and at the same time avoid any scheduling configuration issues. If you think it's not necessary I can remove it.

Reply:
Ah, no, I'm the foremost defender of idempotence 🥇 and definitely wouldn't want you to remove the deletion.
I'm quite happy with your approach here and literally just meant to ask you to please elaborate more on it within the comment in the code (or via a link to a Google doc / wiki page that describes it). In terms of why, my thoughts went to possibly changing the specific implementation of idempotence here later on, by perhaps amending only new/modified lines, or writing into a new table and renaming afterwards - and to make that easier, it'd be useful to have this idempotence guarantee specified somewhere.

Your comment above is quite good and I think you could paste it into the comment almost verbatim - would that be cool?

Reply:
Yeah, sure. :+1: What you are saying makes sense as well though. I will add my comment to the code for better understanding.